app/localization/src/Translation.js

```diff
@@ -61,25 +65,35 @@ export default class Translation extends React.Component<Props, State> {
     this.setTranslatedString();
   };
 
+  componentWillUnmount = () => {
+    if (this.translatePromise !== null) {
+      this.translatePromise.cancel();
+    }
```

Comment:
Did you consider this pattern? https://github.com/kiwicom/mobile/blob/8275d9d0f2eec3d2c70cb0ea78a040dfe07eafa0/app/shared/src/view/AdaptableLayout.js#L55 (I think it's little bit more clean)